bgpd/bgpd.c

```diff
@@ -1398,11 +1401,18 @@ struct peer *peer_new(struct bgp *bgp)
 	 */
 	peer->obuf_work =
 		stream_new(BGP_MAX_PACKET_SIZE + BGP_MAX_PACKET_SIZE_OVERFLOW);
-	peer->ibuf_work =
-		ringbuf_new(BGP_MAX_PACKET_SIZE * BGP_READ_PACKET_MAX);
+	peer->rpkt_quanta =
+		atomic_load_explicit(&bgp->rpkt_quanta, memory_order_relaxed);
 
 	peer->scratch = stream_new(BGP_MAX_PACKET_SIZE);
+	if (peer->rpkt_quanta) {
+		peer->ibuf_work = ringbuf_new(BGP_MAX_PACKET_SIZE * peer->rpkt_quanta);
 
+		peer->ibuf_scratch =
+			XCALLOC(MTYPE_BGP_IBUF_SCRATCH,
+				BGP_EXTENDED_MESSAGE_MAX_PACKET_SIZE
```

Comment:
No. Do not use this macro. Use `BGP_MAX_PACKET_SIZE`.

Reply:
ok